### PR TITLE
guide: (better) SSH remotes

### LINKED ISSUE #3832: guide: working with SSH remote storage
### Report

See details in https://discuss.dvc.org/t/how-do-i-use-dvc-with-ssh-remote/279 (top forum question)

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Closes #3832

In review app: https://dvc-org-guide-remote-ss-w3dcke.herokuapp.com/doc/user-guide/data-management/remote-storage/ssh

---

- [ ] Consider pinning this to https://discuss.dvc.org/t/how-do-i-use-dvc-with-ssh-remote/279/2 when merged?

### CODE REVIEW

File: content/docs/user-guide/data-management/remote-storage/amazon-s3.md [hunk 1]
```diff
@@ -58,8 +58,8 @@ in your project directory locally.
 
 ## Custom authentication
 
-If you don't have the AWS CLI configured in your machine or if you want to
-change the auth method for some reason.
+Use these configuration options if you don't have the AWS CLI setup in your
+environment, if you want to override those values, or to change the auth method.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/data-management/remote-storage/amazon-s3.md:62]: Unrelated change, sorry.

File: content/docs/user-guide/data-management/remote-storage/ssh.md [hunk 2]
```diff
@@ -1,117 +1,117 @@
...
-  2. Port defined in the URL (e.g. `ssh://example.com:1234/path`);
-  3. Port defined in the SSH config file (e.g. `~/.ssh/config`) for this host
-     (URL);
-  4. Default SSH port 22
+<admon type="warn">
 
-- `keyfile` - path to private key to access the remote.
+The `dvc remote modify --local` flag is needed to write sensitive user info to a
+Git-ignored config file (`.dvc/config.local`) so that no secrets are leaked
+through Git. See `dvc config`.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/data-management/remote-storage/ssh.md:71]: > See `dvc config`.

Should replace for config guide when #4379 is merged (here and in other guides where it's already mentioned).

File: content/docs/user-guide/data-management/remote-storage/ssh.md [hunk 3]
```diff
@@ -1,117 +1,117 @@
...
 
-  ```cli
-  $ dvc remote modify --local myremote keyfile /path/to/keyfile
-  ```
+</admon>
 
-- `password` - a password to access the remote
+Using a private key is usually the recommended way to auth an SSH connection,
+and it should be saved in a key file. You can set it's path as shown below.
```
Comment (dberenbaum) [inline content/docs/user-guide/data-management/remote-storage/ssh.md:76]:
```suggestion
Using a private key is usually the recommended way to authenticate an SSH connection,
and it should be saved in a key file. You can set its path as shown below.
```

**Review by dberenbaum — APPROVED** [review]

**Comment by jorgeorpinel** [comment]
> - [ ]  Consider pinning this to https://discuss.dvc.org/t/how-do-i-use-dvc-with-ssh-remote/279/2 when merged?

(Reminder)
